lib/ts/recipe/multifactorauth/index.ts

```diff
@@ -0,0 +1,154 @@
+        const requirements = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getMFARequirementsForAuth({
+            user,
+            accessTokenPayload: session.getAccessTokenPayload(),
+            tenantId: session.getTenantId(),
+            factorsSetUpForUser: factorsSetup,
+            requiredSecondaryFactorsForUser: defaultMFARequirementsForUser,
+            requiredSecondaryFactorsForTenant: defaultMFARequirementsForTenant,
+            completedFactors,
+            userContext: ctx,
+        });
+        Recipe.getInstanceOrThrowError().recipeInterfaceImpl.checkAllowedToSetupFactorElseThrowInvalidClaimError({
+            session,
+            factorId,
+            completedFactors,
+            mfaRequirementsForAuth: requirements,
+            factorsSetUpForUser: factorsSetup,
+            requiredSecondaryFactorsForUser: defaultMFARequirementsForUser,
+            requiredSecondaryFactorsForTenant: defaultMFARequirementsForTenant,
+            userContext: ctx,
+        });
+    }
+
+    static async markFactorAsCompleteInSession(
+        session: SessionContainerInterface,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.markFactorAsCompleteInSession({
+            session,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async getFactorsSetupForUser(userId: string, userContext?: Record<string, any>) {
+        const ctx = getUserContext(userContext);
+        const user = await getUser(userId, ctx);
+        if (!user) {
+            throw new Error("UKNKNOWN_USER_ID");
+        }
+
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getFactorsSetupForUser({
+            user,
+            userContext: ctx,
+        });
+    }
+
+    static async getRequiredSecondaryFactorsForUser(userId: string, userContext?: Record<string, any>) {
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getRequiredSecondaryFactorsForUser({
+            userId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async addToRequiredSecondaryFactorsForUser(
+        userId: string,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.addToRequiredSecondaryFactorsForUser({
+            userId,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async removeFromRequiredSecondaryFactorsForUser(
+        userId: string,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.removeFromRequiredSecondaryFactorsForUser({
+            userId,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+}
+
```

Comment:
Also, make sure all recipe functions are exposed.. getMFARequirementsForAuth is not exposed right now!

Reply:
> we also need to expose all the functions necessary so that users can do all the MFA related checks themselves:
> 
> * getAllAvailableFactorIds
> * getAllAvailableFirstFactorIds
> * addGetFactorsSetupForUserFromOtherRecipes - im not sure if this needs to be exposed.. how would users add their own custom factor?
> * checkForValidFirstFactor
> * checkIfFactorUserCanBeLinkedWithSessionUser
> * linkAccountsForFactorSetup
> * getEmailsForFactors
> * getPhoneNumbersForFactors

I don't think these functions needs to be exposed for user to implement custom factors of their own. These functions are useful only for our login methods. The user should be able to manage to add a custom factor through overrides. I'll try to implement one for my confirmation.

Reply:
> Also, make sure all recipe functions are exposed.. getMFARequirementsForAuth is not exposed right now!

ensured that all recipe interface functions are exposed and easy to use

>

Reply:
So we can expose the functions required for the apis via something like `multifactorauth/helpers`

Reply:
@porcellus